coreclient/src/clients/mod.rs

```diff
@@ -519,9 +521,15 @@ impl CoreUser {
             .ds_send_message(params, group.leaf_signer(), group.group_state_ear_key())
             .await?;
 
-        // Phase 3: Mark the message as sent.
-        let connection = self.connection.lock().await;
+        // Phase 3: Mark the message as sent and read (again).
+        let mut connection = self.connection.lock().await;
         conversation_message.mark_as_sent(&connection, ds_timestamp)?;
+        let mut transaction = connection.transaction()?;
+        Conversation::mark_as_read(
+            &mut transaction,
+            vec![(conversation.id(), conversation_message.id())].into_iter(),
+        )?;
```

Comment:
Why are we marking it as read again here?

Reply:
`conversation_message.mark_as_sent()` correctly updates the timestamp of the message, however, that causes the conversation to appear as unread again. It took me hours to find that bug.